peps/pep-0734.rst

```diff
@@ -0,0 +1,800 @@
+
+* ``create_queue(maxsize=0) -> Queue``
+      Create a new queue.  If the maxsize is zero or negative then the
+      queue is unbounded.
+
+Queue Objects
+-------------
+
+``interpreters.Queue`` objects act as proxies for the underlying
+cross-interpreter-safe queues exposed by the ``interpreters`` module.
+
+``Queue`` implements all the methods of ``queue.Queue`` except for
+``task_done()`` and ``join()``, hence it is similar to
+``asyncio.Queue`` and ``multiprocessing.Queue``.
+
+Attributes and methods:
+
+* ``id``
+      (read-only) The target queue's ``QueueID`` object.
+      It is an int-like object with some internal bookkeeping.
+
+* ``maxsize``
+      Number of items allowed in the queue.
+
+* ``__hash__()``
+      Return the hash of the queue's ``id``.  This is the same
+      as the hash of the ID's integer value.
+
+* ``__eq__(other)``
+      Return ``other is self``.
+
+* ``empty()``
+      Return ``True`` if the queue is empty, ``False`` otherwise.
+
+* ``full()``
+      Return ``True`` if there are ``maxsize`` items in the queue.
+
+      If the queue was initialized with ``maxsize=0`` (the default),
+      then ``full()`` never returns ``True``.
+
+* ``qsize()``
+      Return the number of items in the queue.
+
+* ``put(obj, timeout=None)``
+      Add the object to the queue.
+
+      The object must be `shareable <Shareable Objects_>`_, which means
+      the object's data is passed through rather than the object itself.
+
+      If ``maxsize > 0`` and the queue is full then this blocks until
+      a free slot is available.  If *timeout* is a positive number
+      then it only blocks at least that many seconds and then raises
+      ``interpreters.QueueFull``.  Otherwise is blocks forever.
+
+* ``put_nowait(obj)``
+      Like ``put()`` but effectively with an immediate timeout.
+
+* ``get(timeout=None) -> object``
+      Pop the next object from the queue and return it.  Block while
+      the queue is empty.  If a positive *timeout* is provided and an
+      object hasn't been added to the queue in that many seconds
+      then raise ``interpreters.QueueEmpty``.
+
+* ``get_nowait([default]) -> object``
+      Like ``get()``, but do not block.  If the queue is not empty
+      then return the next item.  Otherwise, return *default* if
+      one was provided else raise ``interpreters.QueueEmpty``.
+
+Shareable Objects
+-----------------
+
+Both ``Interpreter.prepare___main__()`` and ``Queue`` work only with
+"shareable" objects.
+
+A "shareable" object is one which may be passed from one interpreter
+to another.  The object is not necessarily actually shared by the
+interpreters.  However, the object in the one interpreter is guaranteed
+to exactly match the corresponding object in the other interpreter.
+
+For some types (builtin singletons), the actual object is shared.
+For some, the object's underlying data is actually shared but each
+interpreter has a distinct object wrapping that data.  For all other
```

Comment:
I guess that "wrapping" only happens for memoryview?

Reply:
yep